code/modules/mechs/mech_interaction.dm

```diff
@@ -304,8 +304,25 @@
 		P.use(1, user)
 		return TRUE
 
+	else if(istype(I, /obj/item/stack/cable_coil))
+		var/obj/item/stack/cable_coil/coil = I
+		if(coil.amount < 5)
+			to_chat(user, SPAN_WARNING("You need at least 5 cable coil pieces in order to replace wiring."))
+			return TRUE
+		var/obj/item/mech_component/mc = get_targeted_part(user)
+		if(!repairing_check(mc, user))
+			return TRUE
+		if(mc.burn_damage == 0)
+			to_chat(user, SPAN_WARNING("Wiring on this part is already repaired."))
+			return TRUE
+		to_chat(user, SPAN_NOTICE("You start replacing wiring in \the [src]."))
+		if(do_mob(user, src, 30))
+			coil.amount -= 5
+			if(coil.amount == 0)
+				qdel(coil)
```

Comment:
Ah fuck, missed that one. It's a sin to do this manually - coils have `.use(amt)` proc on them for a reason. It's both a check and an update to the amount.

Not to mention that the coil will NOT be deleted if there was only 1 of it, because -4 is != 0.

Reply:
I check before using coils what there is at least 5 pieces or more, but thanks for fix.